samples/wifi/shell/src/wifi_shell.c

```diff
@@ -0,0 +1,341 @@
+ */
+
+/* @file
+ * @brief WPA supplicant shell module
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+
+#include <zephyr/zephyr.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/init.h>
+
+#include <net/net_if.h>
+#include <net/wifi_mgmt.h>
+#include <net/net_event.h>
+#include <zephyr_fmac_main.h>
+
+#ifdef CONFIG_WPA_SUPP
+#include <src/utils/common.h>
+#include <wpa_supplicant/config.h>
+#include <wpa_supplicant/wpa_supplicant_i.h>
+#endif /* CONFIG_WPA_SUPP */
+
+#define SUPPLICANT_SHELL_MODULE "wpa_cli"
+
+static struct {
+	const struct shell *shell;
+
+	union {
+		struct {
+			uint8_t connecting : 1;
+			uint8_t disconnecting : 1;
+			uint8_t _unused : 6;
+		};
+		uint8_t all;
+	};
+} context;
+
+static uint32_t scan_result;
+
+int cli_main(int argc, const char **argv);
+
+extern struct wpa_supplicant *wpa_s_0;
+
+struct wpa_ssid *ssid_0;
+
+#define MAX_SSID_LEN 32
+
+
+static void scan_result_cb(struct net_if *iface,
+			   int status,
+			   struct wifi_scan_result *entry)
+{
+	if (!iface) {
+		return;
+	}
+
+	if (!entry) {
+		if (status) {
+			shell_fprintf(context.shell,
+				      SHELL_WARNING,
+				      "Scan request failed (%d)\n",
+				      status);
+		} else {
+			shell_fprintf(context.shell,
+				      SHELL_NORMAL,
+				      "Scan request done\n");
+		}
+
+		return;
+	}
+
+	scan_result++;
+
+	if (scan_result == 1U) {
+		shell_fprintf(context.shell,
+			      SHELL_NORMAL,
+			      "\n%-4s | %-32s %-5s | %-4s | %-4s | %-5s\n", "Num", "SSID",
+			      "(len)", "Chan", "RSSI", "Sec");
+	}
+
+	shell_fprintf(context.shell,
+		      SHELL_NORMAL,
+		      "%-4d | %-32s %-5u | %-4u | %-4d | %-5s\n",
+		      scan_result,
+		      entry->ssid,
+		      entry->ssid_length,
+		      entry->channel,
+		      entry->rssi,
+		      (entry->security == WIFI_SECURITY_TYPE_PSK ? "WPA/WPA2" : "Open"));
+}
+
+
+static int cmd_supplicant_scan(const struct shell *shell,
+			       size_t argc,
+			       const char *argv[])
+{
+	struct net_if *iface = net_if_get_default();
+	const struct device *dev = net_if_get_device(iface);
+	const struct wifi_nrf_dev_ops *dev_ops = dev->api;
+
+	context.shell = shell;
+
+	return dev_ops->off_api.disp_scan(dev,
+					  scan_result_cb);
+}
+
+
+#ifdef CONFIG_WPA_SUPP
+static int __wifi_args_to_params(size_t argc,
+				 const char *argv[],
+				 struct wifi_connect_req_params *params)
+{
+	char *endptr;
+	int idx = 1;
+
+	if (argc < 1) {
+		return -EINVAL;
+	}
+
+	/* SSID */
+	params->ssid = (char *)argv[0];
+	params->ssid_length = strlen(params->ssid);
+
+	/* PSK (optional) */
+	if (idx < argc) {
+		params->psk = (char *)argv[idx];
+		params->psk_length = strlen(argv[idx]);
+		params->security = WIFI_SECURITY_TYPE_PSK;
+		idx++;
+
+		if ((idx < argc) && (strlen(argv[idx]) <= 2)) {
+			params->security = strtol(argv[idx], &endptr, 10);
+		}
+	} else {
+		params->security = WIFI_SECURITY_TYPE_NONE;
+	}
+
+	return 0;
+}
+
+
+static int cmd_supplicant_connect(const struct shell *shell,
```

Comment:
I would in general replace `cmd_supplicant_` prefix with `cmd_wifi_`. Alternatively, prefix can be removed from static functions.

Reply:
I replace this for scan, but rest of them do go through supplicant, so, left it as is. Hope thats okay.